Lib/test/test_asyncio/test_staggered.py

```diff
@@ -0,0 +1,115 @@
+        self.assertEqual(winner_idx, 0)
+        self.assertEqual(len(exceptions), 1)
+        self.assertIsNone(exceptions[0])
+
+    async def test_single_fail(self):
+        winner_result, winner_idx, exceptions = await staggered_race(
+            (self.get_waiting_coroutine_factory(0, 0.1, False),),
+            0.1,
+        )
+        self.assertEqual(winner_result, None)
+        self.assertEqual(winner_idx, None)
+        self.assertEqual(len(exceptions), 1)
+        self.assertIsInstance(exceptions[0], RuntimeError)
+
+    async def test_first_win(self):
+        winner_result, winner_idx, exceptions = await staggered_race(
+            (
+                self.get_waiting_coroutine_factory(0, 0.2, True),
+                self.get_waiting_coroutine_factory(1, 0.2, True),
+            ),
+            0.1,
+        )
+        self.assertEqual(winner_result, 0)
+        self.assertEqual(winner_idx, 0)
+        self.assertEqual(len(exceptions), 2)
+        self.assertIsNone(exceptions[0])
+        self.assertIsInstance(exceptions[1], asyncio.CancelledError)
+
+    async def test_second_win(self):
+        winner_result, winner_idx, exceptions = await staggered_race(
+            (
+                self.get_waiting_coroutine_factory(0, 0.3, True),
+                self.get_waiting_coroutine_factory(1, 0.1, True),
+            ),
+            0.1,
+        )
+        self.assertEqual(winner_result, 1)
+        self.assertEqual(winner_idx, 1)
+        self.assertEqual(len(exceptions), 2)
+        self.assertIsInstance(exceptions[0], asyncio.CancelledError)
+        self.assertIsNone(exceptions[1])
+
+    async def test_first_fail(self):
+        winner_result, winner_idx, exceptions = await staggered_race(
+            (
+                self.get_waiting_coroutine_factory(0, 0.2, False),
+                self.get_waiting_coroutine_factory(1, 0.2, True),
+            ),
+            0.1,
+        )
+        self.assertEqual(winner_result, 1)
+        self.assertEqual(winner_idx, 1)
+        self.assertEqual(len(exceptions), 2)
+        self.assertIsInstance(exceptions[0], RuntimeError)
+        self.assertIsNone(exceptions[1])
+
+    async def test_second_fail(self):
+        winner_result, winner_idx, exceptions = await staggered_race(
+            (
+                self.get_waiting_coroutine_factory(0, 0.2, True),
+                self.get_waiting_coroutine_factory(1, 0, False),
+            ),
+            0.1,
+        )
+        self.assertEqual(winner_result, 0)
+        self.assertEqual(winner_idx, 0)
+        self.assertEqual(len(exceptions), 2)
+        self.assertIsNone(exceptions[0])
+        self.assertIsInstance(exceptions[1], RuntimeError)
+
+    async def test_simultaneous_success_fail(self):
+        # There's a potential race condition here:
+        # https://github.com/python/cpython/issues/86296
+        # As with any race condition, it can be difficult to reproduce.
+        # This test may not fail every time.
```

Comment:
Please add a note to this comment that the race condition has been fixed in 3.12, e.g.
```suggestion
        # There's a potential race condition here (fixed in Python 3.12):
        # https://github.com/python/cpython/issues/86296
        # As with any race condition, it can be difficult to reproduce.
        # This test may not fail every time.
```